### ISSUE TITLE
Latest version of service clients depends on latest version of core

### ISSUE THREAD

**Issue by lwoggardner** [aws/aws-sdk-ruby#1872]
Since todays release we started getting errors in our builds..

"error	06-Sep-2018 02:01:40	LoadError: cannot load such file -- aws-sdk-core/plugins/client_metrics_plugin.rb"

I'm not entirely sure why our Gemfile.lock was not being respected by bundle install but it was upgrading some of the clients, but not core. 

```
build	06-Sep-2018 02:01:39	Using rake 12.3.1
build	06-Sep-2018 02:01:39	Using aws-sdk-core 3.19.0
build	06-Sep-2018 02:01:39	Using aws-sdk-autoscaling 1.5.0
build	06-Sep-2018 02:01:39	Using aws-sdk-cloudformation 1.4.0
build	06-Sep-2018 02:01:39	Using aws-sdk-cloudwatchlogs 1.6.0 (was 1.3.0)
build	06-Sep-2018 02:01:39	Using aws-sdk-dynamodb 1.12.0 (was 1.5.0)

```

Clearly dynamodb.1.12.0 depends now on core 3.26.0 so that it can require the above file. 
However, the gemspec for dynamodb says -  spec.add_dependency('aws-sdk-core', '~> 3') 

Should it be more specific?

**Comment by BTheunissen** [comment]
I am having the same issue, I had to roll back to an older version of the gem 😞

**Comment by valscion** [comment]
Yeah having the same issue as well. Can't upgrade to latest versions because of this.

**Comment by srchase** [comment]
@lwoggardner 

Thanks for reporting this issue.  We're looking at getting this resolved.

**Comment by awood45** [comment]
I deeply apologize for the mistake here, I'm going to fix this first thing today. A workaround is to hard upgrade `aws-sdk-core` but yes, NEEDING to do that isn't correct. I clearly made a mistake when updating the gemspec generation code.

**Comment by awood45** [comment]
I believe #1871 fixes this as planned, I need to look at the extent of this. Is dynamodb the only affected gem?

**Comment by awood45** [comment]
Okay, this was an oversight on my part. Here are the affected gems:

- aws-sdk-appstream
- aws-sdk-dynamodb
- aws-sdk-elasticloadbalancing
- aws-sdk-rds
- aws-sdk-s3

We'll have new versions of all service gems out today, but I had added plugins to core, yet my update to the gemspec generation logic was put into a separate commit. That was a mistake on my part.

**Comment by awood45** [comment]
Here's a summary:

- I'll post an update here when we push our next release, these gems will be usable with the latest updates after we get that out.
- In the meantime, you can use the latest versions of the current gems by also upgrading `aws-sdk-core` to the latest version. That is essentially what this pull request will do.

**Comment by awood45** [comment]
Update: The release for this is going through our release process now. Expect to be able to update in about 30 minutes from this post.

**Comment by benjamin-cribb** [comment]
This, or something similar, also seems to be effecting aws-sdk-ssm

**Comment by awood45** [comment]
What version of aws-sdk-ssm are you using? Are you seeing the same client error?

**Comment by benjamin-cribb** [comment]
1.24.0, yes, same client error.  Rolling back to aws-sdk-ssm 1.22.0 fixed the problem

**Comment by benjamin-cribb** [comment]
I had to update a bunch of other aws gems to use 1.25, but it also does work

**Comment by benjamin-cribb** [comment]
Try a later version of sns

**Comment by makeitgo** [comment]
Thanks, I was able to update my bundle.  I had to upgrade my s3 which updated my core ....  `bundle update aws-sdk-s3` We had another dependency that was locked to an earlier version.  The error is not gone.
